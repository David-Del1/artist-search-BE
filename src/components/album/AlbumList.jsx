import React, { useState } from 'react';
import { useParams } from 'react-router-dom';
import { useAlbums } from '../../state/artist';
import Album from './Album';
import styled from 'styled-components';

function AlbumList() {
  const [page, setPage] = useState(1);
  const { id } = useParams();

  const { releases, loading } = useAlbums(id, page);
  if(loading) return <h1>Loading...</h1>;

  const albumElements = releases.map(album => (
    <li key={album.id}>
      <Album {...album} />
    </li>
  ));

  return (
    <>
      <AlbumListStyled>
        {albumElements}
      </AlbumListStyled>
      <div className="buttons">
        <button disabled={page <= 1} onClick={() => setPage((prevPage) => prevPage - 1)}>☜</button>
        <button disabled={releases.length < 5} onClick={() => setPage((prevPage) => prevPage + 1)}>☞</button>
      </div>
    </>
  );
}

export default AlbumList;


const AlbumListStyled = styled.ul`
  margin-top: 4rem;
  padding: 1rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  img{
    max-height: 200px;
  }
`;
