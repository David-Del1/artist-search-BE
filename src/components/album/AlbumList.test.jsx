import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import AlbumList from './AlbumList';

describe('Album List Component', () => {
  it('ul and two buttons', () => {
    const album = [{
      'id': '1',
      'title': 'something',
      'url': 'https://supercool.com'
    }];

    const { asFragment } = render (
      <MemoryRouter>
        <AlbumList {...album} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
