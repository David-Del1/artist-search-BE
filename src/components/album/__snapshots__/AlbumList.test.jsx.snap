// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album List Component ul and two buttons 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
